File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int max=0;
        int ans=0;
        for(int i=n-1;i>=0;i--)
        {
            max=Math.max(prices[i], max);
            ans=Math.max(ans, max-prices[i]);
        }
        return ans;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        n=len(prices)
        mx=0
        ans=0
        for i in range(n-1,-1,-1):
            mx=max(prices[i],mx)
            ans=max(ans,mx-prices[i])
        return ans
        
